Objective-C/Tests/RemoteDatabaseTest.m

```diff
@@ -0,0 +1,235 @@
+    config.disableTLS = YES;
+    [self listen: config errorCode: 0 errorDomain: nil];
+    
+    // start the connected client
+    [self startConnectedClient: _listener.localEndpoint.url];
+    
+    [_client deleteDocument: doc1 completion:^(CBLDocument* doc, NSError *error) {
+        AssertNil(error);       // make sure no error
+        AssertNil(doc);         // make sure it 'nil'
+        [e fulfill];
+    }];
+    
+    [self waitForExpectations: @[e] timeout: timeout];
+    
+    [self validateDocument: doc1 errorCode: CBLErrorNotFound];
+}
+
+- (void) testSaveUpdatedDocument {
+    // ---
+    // CREATE A DOC & SYNC & GET IT BACK
+    // ---
+    
+    // save a doc in otherDB
+    NSError* err = nil;
+    CBLMutableDocument* doc1 = [self createDocument: @"doc-1"];
+    [doc1 setString: @"someString" forKey: @"someKeyString"];
+    Assert([self.otherDB saveDocument: doc1 error: &err], @"Fail to save db1 %@", err);
+    
+    // start the listener & get it to 'doc'
+    __block CBLDocument* doc = nil;
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB];
+    config.disableTLS = YES;
+    [self listen: config errorCode: 0 errorDomain: nil];
+    [self startConnectedClient: _listener.localEndpoint.url];
+    XCTestExpectation* eGet = [self expectationWithDescription: @"get document exp"];
+    [_client documentWithID: @"doc-1" completion: ^(CBLDocument* d, NSError* error) {
+        AssertNil(error);                     // empty error
+        doc = d;
+        [eGet fulfill];
+    }];
+    [self waitForExpectations: @[eGet] timeout: timeout];
+    
+    // ---
+    // UPDATE THE DOC
+    // ---
+    XCTestExpectation* eSave = [self expectationWithDescription: @"save document exp"];
+    CBLMutableDocument* doc2 = [doc toMutable];
+    [doc2 setString: @"updated" forKey: @"revised"];
+    [_client saveDocument: doc2
+               completion:^(CBLDocument* d, NSError *error) {
+        AssertNil(error);
+        AssertEqualObjects(d.id, @"doc-1");
+        AssertEqualObjects([d stringForKey: @"someKeyString"], @"someString");
+        AssertEqualObjects([d stringForKey: @"revised"], @"updated");
+        AssertEqual(d.count, 2);
+        [eSave fulfill];
+    }];
+    
+    [self waitForExpectations: @[eSave] timeout: timeout];
+    
+    // ---
+    // GET THE DOC AGAIN AND VERIFY
+    // ---
+    XCTestExpectation* eGet2 = [self expectationWithDescription: @"get document exp2"];
+    [_client documentWithID: @"doc-1" completion: ^(CBLDocument* d, NSError* error) {
+        AssertNil(error);                     // empty error
+        doc = d;
+        [eGet2 fulfill];
+    }];
+    [self waitForExpectations: @[eGet2] timeout: timeout];
+    AssertEqualObjects(doc.id, @"doc-1");
+    AssertEqualObjects([doc stringForKey: @"someKeyString"], @"someString");
+    AssertEqualObjects([doc stringForKey: @"revised"], @"updated");
+    AssertEqual(doc.count, 2);
```

Comment:
Can be simplify with AssertEqualObjects([doc toDictionary], @{.....}).
Validate revID as well.